File: 2998-count-symmetric-integers/2998-count-symmetric-integers.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        symmetric_count = 0
        for num in range(low, high + 1):
            if self.is_symmetric(num):
                symmetric_count += 1
        return symmetric_count

    def is_symmetric(self, n: int) -> bool:
        s = str(n)
        length = len(s)

        if length % 2 != 0:
            return False

        half_length = length // 2

        sum_first_half = 0
        for i in range(half_length):
            sum_first_half += int(s[i]) 

        sum_second_half = 0
        for i in range(half_length, length):
            sum_second_half += int(s[i])

        return sum_first_half == sum_second_half

File: 2998-count-symmetric-integers/2998-count-symmetric-integers.rs
impl Solution {
    pub fn count_symmetric_integers(low: i32, high: i32) -> i32 {
        let mut count = 0;
        for x in low..=high {
            if x < 10 {
                continue;
            } else if x < 100 {
                let a = x / 10;
                let b = x % 10;
                if a == b {
                    count += 1;
                }
            } else if x < 1000 {
                continue;
            } else {
                let a = x / 1000;
                let b = (x / 100) % 10;
                let c = (x / 10) % 10;
                let d = x % 10;
                if a + b == c + d {
                    count += 1;
                }
            }
        }
        count
    }
}